Lib/idlelib/idle_test/test_debugger.py

```diff
@@ -141,65 +133,64 @@ def test_in_rpc_code_debugger_star_dot_py(self):
             test_frame2 = MockFrameType(code_obj, 2)
             test_frame2.f_back = test_frame
 
-            gui = mock.Mock()
-            gui.interaction = mock.Mock()
-
-            idb = debugger.Idb(gui)
+            self.assertFalse(self.idb.in_rpc_code(test_frame2))
 
-            self.assertFalse(idb.in_rpc_code(test_frame2))
 
+class DebuggerTest(unittest.TestCase):
+    """Tests for the idlelib.debugger.Debugger class."""
 
-def make_pyshell_mock():
-    """Factory for generating test fixtures of PyShell."""
-    pyshell = mock.Mock()
-    pyshell.root = None
-    return pyshell
+    @classmethod
+    def setUpClass(cls):
+        cls.root = Tk()
+        cls.root.withdraw()
 
+    @classmethod
+    def tearDownClass(cls):
+        cls.root.destroy()
+        del cls.root
 
-class DebuggerTest(unittest.TestCase):
-    """Tests for the idlelib.debugger.Debugger class."""
+    def setUp(self):
+        self.pyshell = mock.Mock()
+        self.pyshell.root = self.root
+        self.idb = mock.Mock()
+        self.debugger = debugger.Debugger(self.pyshell, self.idb)
+        self.debugger.root = self.root
 
     def test_setup_debugger(self):
         """Test that Debugger can be instantiated with a mock PyShell."""
-        pyshell = make_pyshell_mock()
-        test_debugger = debugger.Debugger(pyshell)
+        test_debugger = debugger.Debugger(self.pyshell)
 
-        self.assertEquals(test_debugger.pyshell, pyshell)
+        self.assertEqual(test_debugger.pyshell, self.pyshell)
         self.assertIsNone(test_debugger.frame)
 
     def test_run_debugger_with_idb(self):
         """Test Debugger.run() with an Idb instance."""
-        mock_idb = mock.Mock()  # Mocked debugger.Idb
-        test_debugger = debugger.Debugger(make_pyshell_mock(), idb=mock_idb)
+        test_debugger = debugger.Debugger(self.pyshell, idb=self.idb)
         test_debugger.run(1, 'two')
-        mock_idb.run.assert_called_once()
-        mock_idb.run.called_with(1, 'two')
-        self.assertEquals(test_debugger.interacting, 0)
+        self.idb.run.assert_called_once()
+        self.idb.run.called_with(1, 'two')
+        self.assertEqual(test_debugger.interacting, 0)
```

Comment:
Since this test uses as IDB instance instead of the default of None, just testing the `self.debugger` wouldn't work.  However, if you made another test class (for example, `DebuggerIDBTest` or something similar), then `DebuggerTest` can do all the tests without an IDB and the other one can do all the tests with an IDB.  That way, you would just be able to use `self.debugger` for all the tests that apply to it.